3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -22,18 +22,100 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error('Something unexpected happened. Please try again later.');
+  }
+  return response.json();
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchAndPopulatePokemons() {
+  const apiUrl = 'https://pokeapi.co/api/v2/pokemon?limit=151';
+  const pokemonData = await fetchData(apiUrl);
+
+  const pokemonDropDownList = createAndAppend(
+    'select',
+    '.container',
+    'pokemon-drop-down-list'
+  );
+  pokemonData.results.forEach((pokemon) => {
+    const { name, url } = pokemon;
+    const optionPokemon = createAndAppend(
+      'option',
+      '.pokemon-drop-down-list',
+      'pokemon-drop-down-list-item'
+    );
+    optionPokemon.textContent = name;
+    optionPokemon.value = url;
+  });
+  pokemonDropDownList.addEventListener('change', fetchImage);
+}
+
+async function fetchImage(e) {
+  if (document.querySelector('.pokemon-img')) {
+    document.querySelector('.pokemon-img').remove();
+  }
+
+  try {
+    const dropdownList = e.target;
+    const { sprites } = await fetchData(dropdownList.value);
```

Comment:
Make sure that your variable names reflect precisely what value they hold, in the case a URL. You can also grab the pokemon name from the returned data for use in the `.alt` attribute.

```suggestion
    const url= e.target.value
    const { sprites, name } = await fetchData(url);
```